typer/core.py

```diff
@@ -738,3 +738,9 @@ def format_help(self, ctx: click.Context, formatter: click.HelpFormatter) -> Non
             ctx=ctx,
             markup_mode=self.rich_markup_mode,
         )
+
+    def list_commands(self, ctx: click.Context) -> List[str]:
+        """Returns a list of subcommand names.
+        Note that in Click's Group class, these are sorted.
+        In Typer, we wish to maintain the original order of creation (cf Issue #933)"""
+        return [n for n, c in self.commands.items()]
```

Comment:
Note that we can't just 
```
return sorted(self.commands)
```
here as `self.commands` is actually a `MutableMapping[str, Command]` and so this wouldn't pass type checks.